src/js/stores/draganddrop.js

```diff
@@ -169,7 +207,7 @@ define(function (require, exports, module) {
                 }.bind(this));
             }
 
-            if ((potentialDropTarget && potentialDropTarget.validate(this._dragTarget, point, potentialDropTarget.b))) {
+            if ((potentialDropTarget && potentialDropTarget.validate(this._dragTarget, point, this._currentBounds))) {
```

Comment:
You don't think number 1 will be worthwhile?

Also, for the record, I'm somewhat _more_ concerned about the regression in `layers.select` than in the drag-and-drop redraw speed because the former is one of the most frequently used operations in the application. Of course, ideally both operations would get faster :)